akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1368,6 +1368,30 @@ final class Flow[In, Out, Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends Gr
   def mapError(pf: PartialFunction[Throwable, Throwable]): javadsl.Flow[In, Out, Mat] =
     new Flow(delegate.mapError(pf))
 
+  /**
+   * While similar to [[recover]] this operator can be used to transform an error signal to a different one *without* logging
+   * it as an error in the process. So in that sense it is NOT exactly equivalent to `recover(t => throw t2)` since recover
+   * would log the `t2` error.
+   *
+   * Since the underlying failure signal onError arrives out-of-band, it might jump over existing elements.
+   * This operator can recover the failure signal, but not the skipped elements, which will be dropped.
+   *
+   * Similarly to [[recover]] throwing an exception inside `mapError` _will_ be logged.
+   *
+   * '''Emits when''' element is available from the upstream or upstream is failed and pf returns an element
+   *
+   * '''Backpressures when''' downstream backpressures
+   *
+   * '''Completes when''' upstream completes or upstream failed with exception pf can handle
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   */
+  def mapError[E <: Throwable](clazz: Class[E], f: function.Function[E, Throwable]): javadsl.Flow[In, Out, Mat] =
```

Comment:
same should also be added to javadsl/Source.scala, javadsl/SubSource.scala, and javadsl/SubFlow.scala

Reply:
sure, i can add it to those three as well. however, it would be unfortunate to just copy/paste the same code. is that really the preferred approach? also, would the unit tests just be a copy/paste of the ones i already wrote in FlowTest?

Reply:
I'm afraid there is no way around that. Creating a subsource and a subflow is different from creating a regular source so you wouldn't be able to just copy paste tests, but you could probably start out with copying the current tests.